### PR TITLE
DellEMC Z9332f platform system health

### PR DESCRIPTION
**Pull request by aravindmani-1**
**- Why I did it**

- Added support for platform system health daemon in Dell Z9332f platform. 

**- How I did it**

- Added the configuration, API changes to support system health

**- How to verify it**

- Execute "show system-health summary/detail/monitor-list" CLI.
- Check SYSTEM_HEALTH_INFO in stateDB.

**- Which release branch to backport (provide reason below if selected)**

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [x] 202012

**- Description for the changelog**
UT logs: 
[z9332f_system_health_UT.txt](https://github.com/Azure/sonic-buildimage/files/5910142/z9332f_system_health_UT.txt)

**- A picture of a cute animal (not mandatory but encouraged)**

### CODE REVIEW

**Comment by jleveque** [comment]
Please fix conflicts

**Comment by aravindmani-1** [comment]
@jleveque Resolved the conflicts.

File: platform/broadcom/sonic-platform-modules-dell/common/sonic_platform/hwaccess.py [hunk 1]
```diff
@@ -22,3 +23,31 @@ def pci_get_value(resource, offset):
 
 def i2c_get(bus, i2caddr, ofs):
     return int(subprocess.check_output(['/usr/sbin/i2cget', '-y', str(bus), str(i2caddr), str(ofs)]), 16)
+
+def io_reg_read(io_resource, offset):
+    fd = os.open(io_resource, os.O_RDONLY)
+    if(fd < 0):
```
Comment (jleveque) [inline platform/broadcom/sonic-platform-modules-dell/common/sonic_platform/hwaccess.py:29]: Remove parentheses

File: platform/broadcom/sonic-platform-modules-dell/common/sonic_platform/hwaccess.py [hunk 2]
```diff
@@ -22,3 +23,31 @@ def pci_get_value(resource, offset):
...
+        print('file open failed %s' % io_resource)
+        return False
+    if(os.lseek(fd, offset, os.SEEK_SET) != offset):
```
Comment (jleveque) [inline platform/broadcom/sonic-platform-modules-dell/common/sonic_platform/hwaccess.py:32]: Remove parentheses

File: platform/broadcom/sonic-platform-modules-dell/common/sonic_platform/hwaccess.py [hunk 3]
```diff
@@ -22,3 +23,31 @@ def pci_get_value(resource, offset):
...
+        print('lseek failed on %s' % io_resource)
+        return False
+    buf = os.read(fd, 1)
+    reg_val1 = ord(buf)
+    os.close(fd)
+    return reg_val1
+
+def io_reg_write(io_resource, offset, val):
+    fd = os.open(io_resource, os.O_RDWR)
+    if(fd < 0):
```
Comment (jleveque) [inline platform/broadcom/sonic-platform-modules-dell/common/sonic_platform/hwaccess.py:42]: Remove parentheses

File: platform/broadcom/sonic-platform-modules-dell/common/sonic_platform/hwaccess.py [hunk 4]
```diff
@@ -22,3 +23,31 @@ def pci_get_value(resource, offset):
...
+        print('file open failed %s' % io_resource)
+        return False
+    if(os.lseek(fd, offset, os.SEEK_SET) != offset):
```
Comment (jleveque) [inline platform/broadcom/sonic-platform-modules-dell/common/sonic_platform/hwaccess.py:45]: Remove parentheses

File: platform/broadcom/sonic-platform-modules-dell/common/sonic_platform/hwaccess.py [hunk 5]
```diff
@@ -22,3 +23,31 @@ def pci_get_value(resource, offset):
...
+        print('lseek failed on %s' % io_resource)
+        return False
+    ret = os.write(fd, struct.pack('B', val))
+    if(ret != 1):
```
Comment (jleveque) [inline platform/broadcom/sonic-platform-modules-dell/common/sonic_platform/hwaccess.py:49]: Remove parentheses

File: platform/broadcom/sonic-platform-modules-dell/z9332f/sonic_platform/chassis.py [hunk 6]
```diff
@@ -317,3 +324,48 @@ def get_reboot_cause(self):
 
     def get_qualified_media_list(self):
         return media_part_num_list
+
+    def initizalize_system_led(self):
+        self.sys_ledcolor = "green"
+
+    def get_status_led(self):
+        val = hwaccess.io_reg_read(self.io_res,self.sysled_offset)
+        if(val != False):
```
Comment (jleveque) [inline platform/broadcom/sonic-platform-modules-dell/z9332f/sonic_platform/chassis.py:333]: Remove parentheses

File: platform/broadcom/sonic-platform-modules-dell/z9332f/sonic_platform/chassis.py [hunk 7]
```diff
@@ -317,3 +324,48 @@ def get_reboot_cause(self):
...
+    def set_status_led(self, color):
+        """ Set system LED status based on the color type passed 
+            in the argument.
+        """
+        
+        if color not in list(self.SYSTEM_LED_COLOR.keys()):
+            raise ValueError("Invalid color type {} provided. Valid colors "
+                   "are {}".format(color,list(self.SYSTEM_LED_COLOR.keys())))
+
+        if(color == "green"):
```
Comment (jleveque) [inline platform/broadcom/sonic-platform-modules-dell/z9332f/sonic_platform/chassis.py:354]: Remove parentheses

File: platform/broadcom/sonic-platform-modules-dell/z9332f/sonic_platform/chassis.py [hunk 8]
```diff
@@ -317,3 +324,48 @@ def get_reboot_cause(self):
...
+            if(not hwaccess.io_reg_write(self.io_res, self.sysled_offset, self.SYSTEM_LED_COLOR[color])):
+                return False
+            self.sys_ledcolor=color
+            return True
+
+        elif(color == "flash_green"):
```
Comment (jleveque) [inline platform/broadcom/sonic-platform-modules-dell/z9332f/sonic_platform/chassis.py:360]: Remove parentheses

File: platform/broadcom/sonic-platform-modules-dell/z9332f/sonic_platform/chassis.py [hunk 9]
```diff
@@ -317,3 +324,48 @@ def get_reboot_cause(self):
...
+            if(not hwaccess.io_reg_write(self.io_res, self.sysled_offset, self.SYSTEM_LED_COLOR[color])):
+                return False
+            self.sys_ledcolor=color
+            return True
+
+        elif(color == "yellow"):
```
Comment (jleveque) [inline platform/broadcom/sonic-platform-modules-dell/z9332f/sonic_platform/chassis.py:366]: Remove parentheses

File: platform/broadcom/sonic-platform-modules-dell/z9332f/sonic_platform/chassis.py [hunk 10]
```diff
@@ -317,3 +324,48 @@ def get_reboot_cause(self):
...
+            if(not hwaccess.io_reg_write(self.io_res, self.sysled_offset, self.SYSTEM_LED_COLOR[color])):
```
Comment (jleveque) [inline platform/broadcom/sonic-platform-modules-dell/z9332f/sonic_platform/chassis.py:367]: Remove parentheses

File: platform/broadcom/sonic-platform-modules-dell/z9332f/sonic_platform/psu.py [hunk 11]
```diff
@@ -21,9 +21,11 @@ class Psu(PsuBase):
 
     # { PSU-ID: { Sensor-Name: Sensor-ID } }
     SENSOR_MAPPING = { 1: { "State": 0x2f, "Current": 0x37,
-                            "Power": 0x38, "Voltage": 0x36 },
+                            "Power": 0x38, "Voltage": 0x36, 
+                            "Temperature": 0x35 },
                        2: { "State": 0x39, "Current": 0x41,
-                            "Power": 0x42, "Voltage": 0x40 } }
+                            "Power": 0x42, "Voltage": 0x40, 
+                             "Temperature": 0x3F } }
```
Comment (jleveque) [inline platform/broadcom/sonic-platform-modules-dell/z9332f/sonic_platform/psu.py:28]: Fix alignment

File: platform/broadcom/sonic-platform-modules-dell/z9332f/sonic_platform/chassis.py [hunk 12]
```diff
@@ -317,3 +324,48 @@ def get_reboot_cause(self):
...
+                return False
+            self.sys_ledcolor=color
```
Comment (jleveque) [inline platform/broadcom/sonic-platform-modules-dell/z9332f/sonic_platform/chassis.py:369]: Add spaces around `=`

File: platform/broadcom/sonic-platform-modules-dell/z9332f/sonic_platform/chassis.py [hunk 13]
```diff
@@ -317,3 +324,48 @@ def get_reboot_cause(self):
...
+        if(color == "green"):
+            if(not hwaccess.io_reg_write(self.io_res, self.sysled_offset, self.SYSTEM_LED_COLOR[color])):
+                return False
+            self.sys_ledcolor=color
+            return True
+
+        elif(color == "flash_green"):
+            if(not hwaccess.io_reg_write(self.io_res, self.sysled_offset, self.SYSTEM_LED_COLOR[color])):
+                return False
+            self.sys_ledcolor=color
```
Comment (jleveque) [inline platform/broadcom/sonic-platform-modules-dell/z9332f/sonic_platform/chassis.py:363]: Add spaces around `=`

File: platform/broadcom/sonic-platform-modules-dell/z9332f/sonic_platform/chassis.py [hunk 14]
```diff
@@ -317,3 +324,48 @@ def get_reboot_cause(self):
...
+        """
+        
+        if color not in list(self.SYSTEM_LED_COLOR.keys()):
+            raise ValueError("Invalid color type {} provided. Valid colors "
+                   "are {}".format(color,list(self.SYSTEM_LED_COLOR.keys())))
+
+        if(color == "green"):
+            if(not hwaccess.io_reg_write(self.io_res, self.sysled_offset, self.SYSTEM_LED_COLOR[color])):
+                return False
+            self.sys_ledcolor=color
```
Comment (jleveque) [inline platform/broadcom/sonic-platform-modules-dell/z9332f/sonic_platform/chassis.py:357]: Add spaces around `=`

File: platform/broadcom/sonic-platform-modules-dell/z9332f/sonic_platform/chassis.py [hunk 15]
```diff
@@ -317,3 +324,48 @@ def get_reboot_cause(self):
...
+
+    def initizalize_system_led(self):
+        self.sys_ledcolor = "green"
+
+    def get_status_led(self):
+        val = hwaccess.io_reg_read(self.io_res,self.sysled_offset)
+        if(val != False):
+            val = hex(val)
+            if val == "0xd0":
+                self.sys_ledcolor="green"
```
Comment (jleveque) [inline platform/broadcom/sonic-platform-modules-dell/z9332f/sonic_platform/chassis.py:336]: Add spaces around `=`

File: platform/broadcom/sonic-platform-modules-dell/z9332f/sonic_platform/chassis.py [hunk 16]
```diff
@@ -317,3 +324,48 @@ def get_reboot_cause(self):
...
+            elif val == "0xe0":
+                self.sys_ledcolor="yellow"
```
Comment (jleveque) [inline platform/broadcom/sonic-platform-modules-dell/z9332f/sonic_platform/chassis.py:338]: Add spaces around `=`

File: platform/broadcom/sonic-platform-modules-dell/z9332f/sonic_platform/chassis.py [hunk 17]
```diff
@@ -317,3 +324,48 @@ def get_reboot_cause(self):
...
+            elif val == "0xd2":
+                self.sys_ledcolor="flash_green"
```
Comment (jleveque) [inline platform/broadcom/sonic-platform-modules-dell/z9332f/sonic_platform/chassis.py:340]: Add spaces around `=`

File: platform/broadcom/sonic-platform-modules-dell/z9332f/sonic_platform/chassis.py [hunk 18]
```diff
@@ -317,3 +324,48 @@ def get_reboot_cause(self):
...
+            else:
+                self.sys_ledcolor="flash_yellow"
```
Comment (jleveque) [inline platform/broadcom/sonic-platform-modules-dell/z9332f/sonic_platform/chassis.py:342]: Add spaces around `=`

**Review by jleveque — CHANGES_REQUESTED** [review]

**Review by jleveque — APPROVED** [review]

**Pull request merged**